src/compas_rhino/__init__.py

```diff
@@ -60,172 +92,223 @@ def _check_rhino_version(version):
-        # However, since it is only used to clean up potential old installations
-        # it does not seem crucial anymore.
-        ironpython_lib_path = None
-        # print("The lib folder for IronPython does not exist in this location: {}".format(ironpython_lib_path))
+def _get_bootstrapper_data(compas_bootstrapper):
+    data = {}
 
-    return ironpython_lib_path
+    if not os.path.exists(compas_bootstrapper):
+        return data
 
+    content = io.open(compas_bootstrapper, encoding='utf8').read()
+    exec(content, data)
 
-def _get_ironpython_lib_path_win32(version):
-    appdata = os.getenv('APPDATA')
-    return os.path.join(appdata,
-                        'McNeel',
-                        'Rhinoceros',
-                        '{}'.format(version),
-                        'Plug-ins',
-                        'IronPython (814d908a-e25c-493d-97e9-ee3861957f49)',
-                        'settings',
-                        'lib')
+    return data
 
 
-def _get_ironpython_lib_path_mac(version):
-    lib_paths = {
-        '5.0': ['/', 'Applications', 'Rhinoceros.app', 'Contents'],
-        '6.0': ['/', 'Applications', 'Rhinoceros.app', 'Contents', 'Frameworks', 'RhCore.framework', 'Versions', 'A'],
-        '7.0': ['/', 'Applications', 'Rhino7.app', 'Contents', 'Frameworks', 'RhCore.framework', 'Versions', 'A'],
-        '8.0': ['/', 'Applications', 'Rhino8.app', 'Contents', 'Frameworks', 'RhCore.framework', 'Versions', 'A'],
-    }
-    return os.path.join(*lib_paths.get(version) + ['Resources', 'ManagedPlugIns', 'RhinoDLR_Python.rhp', 'Lib'])
+def _try_remove_bootstrapper(path):
+    """Try to remove bootstrapper.
 
+    Returns
+    -------
+    bool: ``True`` if the operation did not cause errors, ``False`` otherwise.
+    """
 
-def _get_python_plugins_path(version):
+    bootstrapper = _get_bootstrapper_path(path)
+
+    if os.path.exists(bootstrapper):
+        try:
+            os.remove(bootstrapper)
+            return True
+        except:  # noqa: E722
+            return False
+    return True
+
+
+# =============================================================================
+# Base Application folder (Program Files on Windows and Applications on Mac)
+# =============================================================================
+
+
+def _get_app_folder(version):
     version = _check_rhino_version(version)
+    version = version.split('.')[0]   # take the major only
 
     if compas.WINDOWS:
-        python_plugins_path = _get_python_plugins_path_win32(version)
-
+        app = os.path.join(os.getenv('ProgramFiles'), 'Rhino {}'.format(version))
     elif compas.OSX:
-        python_plugins_path = _get_python_plugins_path_mac(version)
+        paths = {
+            '5': ['/', 'Applications', 'Rhinoceros.app'],
+            '6': ['/', 'Applications', 'Rhinoceros.app'],
+            '7': ['/', 'Applications', 'Rhino 7.app', ],
+            '8': ['/', 'Applications', 'RhinoWIP.app', ]
```

Comment:
I think this part will need a re-work at some point. It probably needs a more elaborate check, like, if we're trying to install on `7`. We could search for `['Rhinoceros.app', `Rhino 7.app`, `Rhino7.app`] and maybe others, and then check in the metadata in that folder to make sure it really is what we want.

Reply:
yes, absolutely. but just to be clear, i haven't changed the functionality (also not this part) just cleaned it up...

Reply:
yep yep, i know... it's just that every time I read it, i remember "uh, we should change this"